cala-server/src/graphql/entry.rs

```diff
@@ -0,0 +1,121 @@
+use async_graphql::{dataloader::DataLoader, types::connection::*, *};
+use serde::{Deserialize, Serialize};
+
+use cala_ledger::primitives::TransactionId;
+
+use super::{
+    convert::ToGlobalId, loader::LedgerDataLoader, primitives::*, transaction::Transaction,
+};
+
+#[derive(SimpleObject)]
+#[graphql(complex)]
+pub struct Entry {
+    id: ID,
+    entry_id: UUID,
+    transaction_id: UUID,
+    journal_id: UUID,
+    account_id: UUID,
+    entry_type: String,
+    sequence: u32,
+    layer: Layer,
+    units: Decimal,
+    currency: CurrencyCode,
+    direction: DebitOrCredit,
+    description: Option<String>,
+    created_at: Timestamp,
+    modified_at: Timestamp,
+}
+
+#[ComplexObject]
+impl Entry {
+    async fn transaction(&self, ctx: &Context<'_>) -> async_graphql::Result<Transaction> {
+        let transaction_id = TransactionId::from(self.transaction_id);
+        let ctx = ctx.data_unchecked::<DataLoader<LedgerDataLoader>>();
+        let transaction = ctx
+            .load_one(transaction_id)
+            .await?
+            .expect("a transaction should always exist for an entry");
```

Comment:
This might actually be `None` for "dummy" entries (like adding / removing accounts to account sets).